tests/components/canary/test_init.py

```diff
@@ -1,73 +1,37 @@
 """The tests for the Canary component."""
-import unittest
+from requests import HTTPError
 
-from homeassistant import setup
-import homeassistant.components.canary as canary
+from homeassistant.components.canary import DOMAIN
+from homeassistant.setup import async_setup_component
 
-from tests.async_mock import MagicMock, PropertyMock, patch
-from tests.common import get_test_home_assistant
+from tests.async_mock import patch
 
 
-def mock_device(device_id, name, is_online=True, device_type_name=None):
-    """Mock Canary Device class."""
-    device = MagicMock()
-    type(device).device_id = PropertyMock(return_value=device_id)
-    type(device).name = PropertyMock(return_value=name)
-    type(device).is_online = PropertyMock(return_value=is_online)
-    type(device).device_type = PropertyMock(
-        return_value={"id": 1, "name": device_type_name}
-    )
-    return device
+async def test_setup_with_valid_config(hass, canary) -> None:
+    """Test setup with valid YAML."""
+    await async_setup_component(hass, "persistent_notification", {})
+    config = {DOMAIN: {"username": "test-username", "password": "test-password"}}
 
+    with patch(
+        "homeassistant.components.canary.alarm_control_panel.setup_platform",
+        return_value=True,
+    ), patch(
+        "homeassistant.components.canary.camera.setup_platform",
+        return_value=True,
+    ), patch(
+        "homeassistant.components.canary.sensor.setup_platform",
+        return_value=True,
+    ):
+        assert await async_setup_component(hass, DOMAIN, config)
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6158285)